lxd/main_init.go

```diff
@@ -494,8 +510,148 @@ they otherwise would.
 			return err
 		}
 	}
+	return nil
+}
+
+// Run the logic for preseed mode
+func (cmd *CmdInit) runPreseed(client lxd.ContainerServer) error {
+	data := &cmdInitData{}
+
+	err := cmd.Context.InputYAML(data)
+	if err != nil {
+		return fmt.Errorf("Invalid preseed YAML content")
+	}
+
+	return cmd.run(client, data)
+}
+
+// Apply the configuration specified in the given init data.
+func (cmd *CmdInit) run(client lxd.ContainerServer, data *cmdInitData) error {
```

Comment:
You can fully delete a storage pool you just created. The corresponding client function is in `client.go` or at the appropriate place for the new client under `client/*`.

It's not transactional in the sense of "create this set of things (network, pools, etc.) and either succeed or leave the old state". There's a bunch of loops in the functions you wrote and each iteration creates or updates a resource. By design that's not transactional. The revert logic is the closest we get to restoring the old state. If you see a better way though, go ahead.